src/main/java/com/demo/MybatisApplication/service/TeacherService.java

```diff
@@ -15,20 +17,30 @@
 public class TeacherService {
     @Autowired
     TeacherRepository teacherRepository;
-
     @Autowired
     TeacherMapper teacherMapper;
 
     public TeacherDisplayDto getTeacherById(@PathVariable long id){
-        return teacherMapper.teacherEntityToTeacherDisplayDto(teacherRepository.findTeacherById(id));
+        TeacherEntity teacherEntity = teacherRepository.findTeacherById(id);
+        if (teacherEntity == null) {
+            throw new RuntimeException("Teacher not found with ID: " + id);
+        }
+        return teacherMapper.mapTeacherEntityToTeacherDisplayDto(teacherEntity);
     }
 
-    public List<TeacherDisplayWithIdDto> getAllTeachers(){
-        List<TeacherDisplayWithIdDto> teacherDisplayWithIdDtos = teacherMapper.teacherDiplayIdDtosFromEntities(teacherRepository.findAllTeachers());
-        return teacherDisplayWithIdDtos;
+    public List<TeacherEntityDisplayDto> getAllTeachers(){
+        List<TeacherEntityDisplayDto> teacherEntityDisplayDtos = teacherMapper.mapTeacherEntitiesToTeacherEntityDisplayDto(teacherRepository.findAllTeachers());
+        return teacherEntityDisplayDtos;
     }
 
-    public Object getTeacherWithSubjectsById(@PathVariable Long id) {
-        return teacherRepository.findSubjectsAssignedToTeacher(id);
+    public ResponseEntity<TeacherEntityDisplayDto> addTeacher(TeacherAdditionDto teacher){
+        if (teacher.getName() == null || teacher.getName().isEmpty()) {
```

Comment:
If we have notBlank annotation then this if case is not needed.

Reply:
Okay